### PR TITLE
Send policy modified message reason

### PR DESCRIPTION
**Pull request by phuongnq**
https://github.com/craftercms/craftercms/issues/4730

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java [hunk 1]
```diff
@@ -58,18 +60,27 @@ private void validatePermitted(HierarchicalConfiguration<?> config, Action actio
         var sourceRegex = config.getString(CONFIG_KEY_SOURCE_REGEX);
         if (!item.matches(sourceRegex)) {
             String modifiedItem = null;
+            StringBuilder reasonBuilder = new StringBuilder();
             var targetRegex = config.getString(CONFIG_KEY_TARGET_REGEX);
             if (targetRegex != null) {
                 modifiedItem = item.replaceAll(sourceRegex, targetRegex);
-
+                if (!modifiedItem.equalsIgnoreCase(item)) {
+                    addPolicyReason(reasonBuilder, format("Path is replaced from source regex '%s' to target regex '%s'.", sourceRegex, targetRegex));
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: Let's change this to show the user the before and after. Something like "Path '%s' was transformed to '%s' per the project file name policy."

Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: The above can be used for all cases. No need for lowercase/uppercase.

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java [hunk 2]
```diff
@@ -58,18 +60,27 @@ private void validatePermitted(HierarchicalConfiguration<?> config, Action actio
...
+                }
                 var caseTransform = config.getString(CONFIG_KEY_CASE_TRANSFORM);
                 if (isNotEmpty(caseTransform)) {
                     switch (caseTransform.toLowerCase()) {
                         case "uppercase":
-                            modifiedItem = modifiedItem.toUpperCase();
+                            if (modifiedItem.chars().anyMatch(c -> Character.isLowerCase(c))) {
+                                addPolicyReason(reasonBuilder, "Path is transformed to uppercase.");
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: Maybe we use the one reason as I mention above.

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java [hunk 3]
```diff
@@ -58,18 +60,27 @@ private void validatePermitted(HierarchicalConfiguration<?> config, Action actio
...
+                                modifiedItem = modifiedItem.toUpperCase();
+                            }
                             break;
                         case "lowercase":
-                            modifiedItem = modifiedItem.toLowerCase();
+                            if (modifiedItem.chars().anyMatch(c -> Character.isUpperCase(c))) {
+                                addPolicyReason(reasonBuilder, "Path is transformed to lowercase.");
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: Maybe we use the one reason as I mention above.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java (hunk 1 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: I updated the reason message.

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: I updated the reason message.

File: src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java (hunk 3 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java]: I updated the reason message.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**